### ISSUE TITLE
Add citar-copy-reference

### ISSUE THREAD

**Issue by bdarcus** [bdarcus/citar#448]
~We could either change the existing command behavior, and probably rename it, or add a new command?~

Let's add `citar-copy-reference` per https://github.com/bdarcus/citar/issues/448#issuecomment-981818707 below, and make that available in the at-point keymap.

See also #245, and in particular https://github.com/bdarcus/citar/issues/245#issuecomment-985480741.

-----------

For me personally, it would probably be easier to use a function that just puts the formatted reference in the kill ring directly.

_Originally posted by @xcapaldi in https://github.com/bdarcus/citar/issues/447#issuecomment-981708298_

**Comment by xcapaldi** [comment]
I think change existing behavior and rename. Maybe something like this:

```
(defcustom citar-get-reference-function
  #'citar--get-reference
  "A function that takes a list of (KEY . ENTRY), and adds the formatted references to the kill ring."
  :group 'citar
  :type 'function)

;;;###autoload
(defun citar-get-reference (keys-entries)
  "Add formatted reference(s) associated with the KEYS-ENTRIES to the kill ring.
With prefix, rebuild the cache before offering candidates."
  (interactive (list (citar-select-refs
                      :rebuild-cache current-prefix-arg)))
  (apply citar-get-reference-function (list keys-entries)))

(defun citar--get-reference (keys-entries)
  "Add formatted reference(s) associated with the KEYS-ENTRIES to the kill ring."
  (let ((template (citar-get-template 'preview)))
    (dolist (key-entry keys-entries)
      (when template
        (kill-new (citar--format-entry-no-widths (cdr key-entry) template))))))
```

I haven't tested what happens when multiple entries are selected at once.

**Comment by localauthor** [comment]
I have both functions called, simply, `insert-reference` and `copy-reference`. I put both in my `citar-map` and just `copy-reference` in my `citar-citation-map`, so I can quickly get a full reference from a key at point (where inserting doesn’t make much sense to me).

**Comment by bdarcus** [comment]
Forgot to link; closed via #455.
